src/components/SourceDetail/SourceSummaryCard.js

```diff
@@ -31,6 +31,17 @@ const SourceSummaryCard = () => {
   const intl = useIntl();
   const source = useSource();
   const sourceTypes = useSelector(({ sources }) => sources.sourceTypes, shallowEqual);
+  const type = sourceTypes.find((type) => type.id === source.source_type_id);
+
+  let awsAccountNumber;
+
+  if (type.name === 'amazon') {
+    const arnType = source.authentications?.find(({ authtype }) => authtype.includes('arn'));
+
+    if (arnType?.username) {
+      awsAccountNumber = arnType.username.match(/:\d+/)?.[0].replace(/:/g, '');
```

Comment:
Maybe we want to add the conditional accessor before the `.replace` as well

Reply:
Hmm, I don't think so, because if there is a match, it will be a string all the time, won't be?

Reply:
Yeah, it should be there. It's just my brain is always warning me about accessing stuff in the array via index :ghost: